scripts/import_from_biosamples.py

```diff
@@ -27,49 +26,102 @@
     'FAANG Legacy Samples': 'Legacy'
 }
 TOTAL_RECORDS_TO_UPDATE = 0
-
-
+ETAGS_CACHE = dict()
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(format='%(asctime)s\t%(levelname)s:\t%(name)s line %(lineno)s\t%(message)s', level=logging.INFO)
+# suppress logging information from elasticsearch
+logging.getLogger('elasticsearch').setLevel(logging.WARNING)
+
+
+@click.command()
+@click.option(
+    '--es_hosts',
+    default="wp-np3-e2;wp-np3-e3",
+    help='Specify the Elastic Search server(s) (port could be included), e.g. wp-np3-e2:9200. '
+         'If multiple servers are provided, please use ";" to separate them, e.g. "wp-np3-e2;wp-np3-e3"'
+)
+@click.option(
+    '--es_index_prefix',
+    default="",
+    help='Specify the Elastic Search index prefix, e.g. '
+         'faang_build_1_ then the indices will be faang_build_1_organism etc.'
+         'If not provided, then work on the aliases'
+)
 # TODO check single or double quotes
-def main():
-    ruleset_version = get_ruleset_version()
-    es = Elasticsearch(['wp-np3-e2', 'wp-np3-e3'])
-
-    print(f"The program starts at {datetime.datetime.now()}")
-    print(f"Current ruleset version is {ruleset_version}")
-
-    etags: Dict[str, str] = get_existing_etags()
+def main(es_hosts, es_index_prefix):
+    global ETAGS_CACHE
+    today = date.today().strftime('%Y-%m-%d')
+    try:
+        with open("etag_list_{}.txt".format(today), 'r') as f:
+            for line in f:
+                line = line.rstrip()
+                data = line.split("\t")
+                ETAGS_CACHE[data[0]] = data[1]
+    except FileNotFoundError:
+        logger.error(f"Could not find the local etag cache file etag_list_{today}.txt")
+        sys.exit(1)
+
+    hosts = es_hosts.split(";")
+    logger.info("Command line parameters")
+    logger.info("Hosts: "+str(hosts))
+    if es_index_prefix:
+        logger.info("Index_prefix:"+es_index_prefix)
 
-    print(f"There are {len(etags)} records with etags in ES")
-    print(f"Finish retrieving existing etags at {datetime.datetime.now()}")
-    print("Importing FAANG data")
+    ruleset_version = get_ruleset_version()
+    es = Elasticsearch(hosts)
+
+#   pool of specimen
```

Comment:
Do we need to have all these comments in a main code? I understand that it is convenient for test purposes, but it really makes this code harder to read and understand, so I think that it would be better to write local script that will do these steps.

Reply:
The main reason for this is to show third party programmer how to debug the codes.

Reply:
The code will be easier to understand with unnecessary parts removed, they will have the test cases from the separate scripts once these are included.

Reply:
still hold the opinion the comment could help to understand the code. 
Anyway it has been removed.